oem/firstvoices/ios/FirstVoices/KeyboardRepository.swift

```diff
@@ -84,41 +84,108 @@ class KeyboardRepository {
+    // EuroLatin SIL is not tracked as an active keyboard in the KeyboardSettingsRepository.
+    // This is intentional as it should go away once FV keyboards are added.
+
+    rebuildKeyboardsListInKeymanEngine(keyboards: newKeyboardsMap)
     
-    if let keyboardsPackage = loadKeyboardPackage() {
-      let fullKeyboardId = FullKeyboardID(keyboardID: keyboard.keyboardId, languageID: keyboard.languageTag)
-      do {
-        FVShared.reportState(location: "before install keyboard")
-        try ResourceFileManager.shared.install(resourceWithID: fullKeyboardId, from: keyboardsPackage)
-        FVShared.reportState(location: "after install keyboard")
-        print("Installed keyboard \(keyboard.keyboardId)")
-        success = true
-      } catch {
-        let installError = error
-        
-        print("Failed to load preload \(keyboard.keyboardId) installError: \(installError.localizedDescription)")
-        success = false
-     }
-    }
+    return true
     
-    return success
   }
 
   func removeKeyboard(keyboard: FVKeyboardDefinition) -> Bool {
-    var success = false
-    let fullKeyboardId = FullKeyboardID(keyboardID: keyboard.keyboardId, languageID: keyboard.languageTag)
-    FVShared.reportState(location: "before remove keyboard")
-    let removed = Manager.shared.removeKeyboard(withFullID: fullKeyboardId)
-    FVShared.reportState(location: "after remove keyboard")
-    if (removed) {
-        print("Removed keyboard \(keyboard.keyboardId)")
-        success = true
-    } else {
-      print("Could not remove keyboard \(keyboard.keyboardId)")
-      success = true
-   }
-    return success
+    // make a copy of all the active keyboards
+    var newKeyboardsMap = Dictionary(uniqueKeysWithValues:
+                                      KeyboardSettingsRepository.shared.activeKeyboards.map { key, value in (key, value.languageTag) })
+    // remove keyboard to be deleted
+    newKeyboardsMap.removeValue(forKey: keyboard.keyboardId)
+    
+    rebuildKeyboardsListInKeymanEngine(keyboards: newKeyboardsMap)
+    
+    return true
+  }
+  
+  func rebuildKeyboardsListInKeymanEngine(keyboards: [String:String]) {
+
+    // call Keyman Engine to remove all keyboards
+    while Manager.shared.removeKeyboard(at: 0) {
+      print("removed existing keyboard")
+    }
+    
+    // re-install specified keyboards 
+    if let keyboardsPackage = loadKeyboardPackage() {
+      for (keyboardId, languageTag) in keyboards {
+        let fullKeyboardId = FullKeyboardID(keyboardID: keyboardId, languageID: languageTag)
+        do {
+          try ResourceFileManager.shared.install(resourceWithID: fullKeyboardId, from: keyboardsPackage)
+          print("Installed keyboard \(keyboardId)")
+        } catch {
+          let installError = error
+          print("Failed to load preload \(keyboardId) installError: \(installError.localizedDescription)")
```

Comment:
Can we make sure we report the error to Sentry?

Reply:
There is a draft PR that Josh created for FV Sentry integration, but nothing goes to Sentry from FV today, and I have not looked into that.

Reply:
Gotcha